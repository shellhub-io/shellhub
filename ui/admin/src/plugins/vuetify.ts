// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
