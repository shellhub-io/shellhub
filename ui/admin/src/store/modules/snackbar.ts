import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
