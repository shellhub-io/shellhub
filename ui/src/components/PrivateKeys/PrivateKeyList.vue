<template>
  <v-table class="bg-background border rounded mx-4">
    <thead class="bg-v-theme-background" data-test="privateKey-thead">
      <tr>
        <th
          v-for="(head, i) in headers"
          :key="i"
          :class="head.align ? `text-${head.align}` : 'text-center'"
        >
          <span> {{ head.text }}</span>
        </th>
      </tr>
    </thead>
    <tbody v-if="privateKeys.length">
      <tr v-for="(privateKey, i) in privateKeys" :key="i">
        <td class="text-center" data-test="privateKey-name">
          {{ privateKey.name }}
        </td>
        <td class="text-center" data-test="privateKey-fingerprint">
          {{ getKeyFingerprint(privateKey) }}
        </td>
        <td class="text-center">
          <v-menu
            location="bottom"
            scrim
            eager
          >
            <template v-slot:activator="{ props }">
              <v-btn
                v-bind="props"
                variant="plain"
                class="border rounded bg-v-theme-background"
                density="comfortable"
                size="default"
                icon="mdi-format-list-bulleted"
                data-test="privateKey-actions"
              />
            </template>
            <v-list class="bg-v-theme-surface" lines="two" density="compact">
              <PrivateKeyEdit
                :private-key="privateKey"
                @update="getPrivateKeys"
              />

              <PrivateKeyDelete
                :id="privateKey.id"
                @update="getPrivateKeys"
              />
            </v-list>
          </v-menu>
        </td>
      </tr>
    </tbody>
    <div v-else sm="12" class="text-start mt-2 mb-3" data-test="no-private-key-warning">
      <span class="ml-4">No data available</span>
    </div>
  </v-table>
</template>

<script setup lang="ts">
import { computed, onMounted } from "vue";
import PrivateKeyDelete from "./PrivateKeyDelete.vue";
import PrivateKeyEdit from "./PrivateKeyEdit.vue";
import handleError from "@/utils/handleError";
import { IPrivateKey } from "@/interfaces/IPrivateKey";
import { convertToFingerprint } from "@/utils/validate";
import usePrivateKeysStore from "@/store/modules/private_keys";

const privateKeysStore = usePrivateKeysStore();

const headers = [
  {
    text: "Name",
    value: "name",
    align: "center",
    sortable: true,
  },
  {
    text: "Fingerprint",
    value: "data",
    align: "center",
    sortable: true,
  },
  {
    text: "Actions",
    value: "actions",
    align: "center",
    sortable: false,
  },
];
const privateKeys = computed(() => privateKeysStore.privateKeys);

const getPrivateKeys = () => {
  try {
    privateKeysStore.getPrivateKeyList();
  } catch (error: unknown) {
    handleError(error);
  }
};

const getKeyFingerprint = (privateKey: IPrivateKey) => {
  if (privateKey.fingerprint) {
    return privateKey.fingerprint;
  }

  const fingerprint = convertToFingerprint(privateKey.data);
  return fingerprint || "Fingerprint not available";
};

onMounted(() => {
  getPrivateKeys();
});
</script>
