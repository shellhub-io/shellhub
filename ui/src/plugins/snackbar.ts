import { App, Plugin, reactive } from "vue";

const SnackbarInjectionKey = Symbol("snackbar");

type SnackbarType = "success" | "error" | "info" | "warning";

interface SnackbarState {
  message: string;
  type: SnackbarType;
  show: boolean;
}

interface ISnackbarPlugin {
  showSuccess: (msg: string) => void;
  showError: (msg: string) => void;
  showInfo: (msg: string) => void;
  showWarning: (msg: string) => void;
  getMessage: () => string;
  getType: () => SnackbarType;
  getShow: () => boolean;
}

const state = reactive<SnackbarState>({
  message: "",
  type: "info",
  show: false,
});

const showSnackbar = (type: SnackbarType, message: string) => {
  state.message = message;
  state.type = type;
  state.show = true;

  setTimeout(() => {
    state.show = false;
  }, 4000);
};

const plugin: ISnackbarPlugin = {
  showSuccess: (msg: string) => showSnackbar("success", msg),
  showError: (msg: string) => showSnackbar("error", msg),
  showInfo: (msg: string) => showSnackbar("info", msg),
  showWarning: (msg: string) => showSnackbar("warning", msg),
  getMessage: () => state.message,
  getType: () => state.type,
  getShow: () => state.show,
};

const SnackbarPlugin: Plugin = {
  install(app: App) {
    app.provide(SnackbarInjectionKey, plugin);
  },
};

export { SnackbarPlugin, SnackbarInjectionKey, plugin };
export type { SnackbarState, ISnackbarPlugin };
