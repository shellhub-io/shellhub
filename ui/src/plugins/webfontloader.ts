// webfontloader documentation: https://github.com/typekit/webfontloader
const loadFonts = async () => {
  const webFontLoader = await import("webfontloader");

  webFontLoader.load({
    google: {
      families: ["Roboto:100,300,400,500,700,900&display=swap"],
    },
  });
};

export default loadFonts;
