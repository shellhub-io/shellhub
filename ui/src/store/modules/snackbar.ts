import { Module } from "vuex";
import { State } from "..";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  message: string,
  type: string,
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,
  state: {
    message: "",
    type: "",
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: {
      typeMessage: "",
      typeContent: "",
    },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    showMessage: (state, data) => {
      state.type = data.type;
      state.message = data.message;
    },

    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarSuccessNotRequest: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = {
        typeMessage: "default",
        typeContent: "",
      };
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = {
        typeMessage: "",
        typeContent: value,
      };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = {
        typeMessage: "action",
        typeContent: value,
      };
      commit("setSnackbarSuccessAction", data);
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarSuccessNotRequest: (context, value) => {
      const data = {
        typeMessage: "notRequest",
        typeContent: value,
      };
      context.commit("setSnackbarSuccessNotRequest", data);
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = {
        typeMessage: "loading",
        typeContent: value,
      };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = {
        typeMessage: "action",
        typeContent: value,
      };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
