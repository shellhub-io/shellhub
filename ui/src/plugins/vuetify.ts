// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
