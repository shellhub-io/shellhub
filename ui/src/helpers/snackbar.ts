import { inject } from "vue";
import { InjectionKey, type ISnackbarPlugin } from "@/plugins/snackbar";

const useSnackbar = (): ISnackbarPlugin => inject(InjectionKey) as ISnackbarPlugin;

export default useSnackbar;
