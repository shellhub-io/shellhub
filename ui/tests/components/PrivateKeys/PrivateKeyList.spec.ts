import { setActivePinia, createPinia } from "pinia";
import { beforeEach, describe, expect, it } from "vitest";
import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import PrivateKeyList from "@/components/PrivateKeys/PrivateKeyList.vue";
import { SnackbarPlugin } from "@/plugins/snackbar";
import usePrivateKeysStore from "@/store/modules/private_keys";

type PrivateKeyListWrapper = VueWrapper<InstanceType<typeof PrivateKeyList>>;

const mockPrivateKeys = [
  { id: 1, name: "test-key-1", data: "private-key-data-1", hasPassphrase: true, fingerprint: "fingerprint-1" },
  { id: 2, name: "test-key-2", data: "private-key-data-2", hasPassphrase: false, fingerprint: "fingerprint-2" },
  { id: 3, name: "test-key-3", data: "private-key-data-3", hasPassphrase: false, fingerprint: "fingerprint-3" },
];

describe("Private Key List", () => {
  let wrapper: PrivateKeyListWrapper;
  setActivePinia(createPinia());
  const privateKeysStore = usePrivateKeysStore();
  const vuetify = createVuetify();

  beforeEach(async () => {
    wrapper = mount(PrivateKeyList, {
      global: {
        plugins: [vuetify, SnackbarPlugin],
      },
    });

    privateKeysStore.privateKeys = mockPrivateKeys;
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
