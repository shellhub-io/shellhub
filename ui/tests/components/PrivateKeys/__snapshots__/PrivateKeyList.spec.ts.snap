// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Private Key List > Renders the component 1`] = `
"<div class="v-table v-theme--light v-table--density-default bg-background border rounded mx-4">
  <!---->
  <div class="v-table__wrapper">
    <table>
      <thead class="bg-v-theme-background" data-test="privateKey-thead">
        <tr>
          <th class="text-center"><span>Name</span></th>
          <th class="text-center"><span>Fingerprint</span></th>
          <th class="text-center"><span>Actions</span></th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td class="text-center" data-test="privateKey-name">test-key-1</td>
          <td class="text-center" data-test="privateKey-fingerprint">fingerprint-1</td>
          <td class="text-center"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--size-default v-btn--variant-plain border rounded bg-v-theme-background" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-0" data-test="privateKey-actions"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-format-list-bulleted mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <!--teleport start-->
            <!--teleport end-->
          </td>
        </tr>
        <tr>
          <td class="text-center" data-test="privateKey-name">test-key-2</td>
          <td class="text-center" data-test="privateKey-fingerprint">fingerprint-2</td>
          <td class="text-center"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--size-default v-btn--variant-plain border rounded bg-v-theme-background" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-2" data-test="privateKey-actions"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-format-list-bulleted mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <!--teleport start-->
            <!--teleport end-->
          </td>
        </tr>
        <tr>
          <td class="text-center" data-test="privateKey-name">test-key-3</td>
          <td class="text-center" data-test="privateKey-fingerprint">fingerprint-3</td>
          <td class="text-center"><button type="button" class="v-btn v-btn--icon v-theme--light v-btn--density-comfortable v-btn--size-default v-btn--variant-plain border rounded bg-v-theme-background" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-v-4" data-test="privateKey-actions"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
              <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-format-list-bulleted mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
              <!---->
              <!---->
            </button>
            <!--teleport start-->
            <!--teleport end-->
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <!---->
</div>"
`;
